File: krojac-ui/src/app/model/curtainRequest.ts
export class CurtainRequest {
  height!: number;       // window height
  width!: number;        // window width
  pleatFactor!: number;  // 1.5, 2, etc.
  hangingType!: string;  // HOOKS, RINGS, CHANNEL
  marginTop!: number;   // top margin
  marginBottom!: number; // bottom margin

  constructor(init?: Partial<CurtainRequest>) {
    Object.assign(this, init);
  }
}

File: sbnz-integracija-projekta/service/src/main/java/com/ftn/sbnz/service/dto/CurtainRequest.java
package com.ftn.sbnz.service.dto;

public class CurtainRequest {
    private double height;
    private double width;
    private double pleatFactor;
    private String hangingType; // HOOKS, RINGS, CHANNEL
    private double marginTop;
    private double marginBottom;

    public double getHeight() {
        return height;
    }

    public void setHeight(double height) {
        this.height = height;
    }

    public double getWidth() {
        return width;
    }

    public void setWidth(double width) {
        this.width = width;
    }

    public double getPleatFactor() {
        return pleatFactor;
    }

    public void setPleatFactor(double pleatFactor) {
        this.pleatFactor = pleatFactor;
    }

    public String getHangingType() {
        return hangingType;
    }

    public void setHangingType(String hangingType) {
        this.hangingType = hangingType;
    }

    public double getMarginTop() {
        return marginTop;
    }

    public void setMarginTop(double marginTop) {
        this.marginTop = marginTop;
    }

    public double getMarginBottom() {
        return marginBottom;
    }

    public void setMarginBottom(double marginBottom) {
        this.marginBottom = marginBottom;
    }
}
